### PR TITLE
Use snake_case `code_examples` like every other folder we have

### PR DESCRIPTION
**Pull request by emilk**
### What
So irritating when the folder and crate name is different _for one single crate_

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3596) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3596)
- [Docs preview](https://rerun.io/preview/3edfcb332cf8894f45115764548986668f1f62e8/docs) 
- [Examples preview](https://rerun.io/preview/3edfcb332cf8894f45115764548986668f1f62e8/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Review by Wumpf — CHANGES_REQUESTED** [review]
Breaks doc page right now:

Before: https://www.rerun.io/preview/69900eaacb6d71faf79e49dfeb84cf33ccc1f673/docs/reference/data_types/archetypes/arrows3d
After: https://www.rerun.io/preview/3edfcb332cf8894f45115764548986668f1f62e8/docs/reference/data_types/archetypes/arrows3d

**Comment by Wumpf** [comment]
maybe this requires a change on `landing`? In that case, maybe this is a bad moment for this refactor

File: CHANGELOG.md [hunk 1]
```diff
@@ -185,7 +185,7 @@ for use-cases like real-time video feeds. [#2220](https://github.com/rerun-io/re
 - Added example screenshot instructions in `just upload --help` [#2454](https://github.com/rerun-io/rerun/pull/2454)
 - Added support for puling image from an URL to `upload_image.py` [#2462](https://github.com/rerun-io/rerun/pull/2462)
 - `setup_dev.sh` now installs pngcrush [#2470](https://github.com/rerun-io/rerun/pull/2470)
-- Added docs/code-examples to the directories checked by py-lint and py-format [#2476](https://github.com/rerun-io/rerun/pull/2476)
+- Added docs/code_examples to the directories checked by py-lint and py-format [#2476](https://github.com/rerun-io/rerun/pull/2476)
```
Comment (Wumpf) [inline CHANGELOG.md:188]: let's not change the changelog retroactively!

**Comment by abey79** [comment]
Worth noting that all docs folders currently use kebab case.

**Comment by jprochazk** [comment]
> maybe this requires a change on landing?

Yes, it's hardcoded in a bunch of places there: https://github.com/rerun-io/landing/blob/main/src/github.ts#L177

**Pull request closed without merging**